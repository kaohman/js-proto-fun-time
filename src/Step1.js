import React from 'react';

function Step1(props) {
  return (
    <div className='step-containers' id='step-1-container'>
      <div>
        <h3>Step 1: Read The Problem</h3>
        <p className='question-text'>{props.question}</p>
      </div>
      {
        props.currentStep === 1 && 
        <button onClick={props.incrementStep}>Next Step</button>
      }
    </div>
  )
}

export default Step1;