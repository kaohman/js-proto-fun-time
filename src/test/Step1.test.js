import React from 'react';
import ReactDOM from 'react-dom';
import Step1 from '../Step1';
import { shallow } from 'enzyme';

let incrementStepMock = jest.fn();
let questionMock = 'You are given the following array of your friends ages: [18, 29, 56, 5, 77, 1]. You have a wide variety or friends! All your friends have celebrated their birthdays this year so you want to return an updated array of ages that reflects them all aging by one year.'
let currentStepMock = 1

describe('Step1', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallow(
      <Step1 
        incrementStep={incrementStepMock}
        question={questionMock}
        currentStep={currentStepMock}
      />
    );
  });

  it('should match the snapshot with all data passed in', () => {
    expect(wrapper).toMatchSnapshot();
  });
});