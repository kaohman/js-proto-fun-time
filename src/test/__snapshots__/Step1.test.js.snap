// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Step1 should match the snapshot with all data passed in 1`] = `
<div
  className="step-containers"
  id="step-1-container"
>
  <div>
    <h3>
      Step 1: Read The Problem
    </h3>
    <p>
      You are given the following array of your friends ages: [18, 29, 56, 5, 77, 1]. You have a wide variety or friends! All your friends have celebrated their birthdays this year so you want to return an updated array of ages that reflects them all aging by one year.
    </p>
  </div>
  <button
    onClick={[MockFunction]}
  >
    Next Step
  </button>
</div>
`;
